src/Features/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -154,5 +155,14 @@ public static async Task<NamingRule> GetApplicableNamingRuleAsync(this Document
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        public static ImmutableArray<AbstractFormattingRule> GetFormattingRules(this Document document, IEnumerable<AbstractFormattingRule>? rules, TextSpan span)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var formattingRuleFactory = workspace.Services.GetRequiredService<IHostDependentFormattingRuleFactoryService>();
+            var position = (span.Start + span.End) / 2;
```

Comment:
doc..

Reply:
I dug into this, it looks like it's venus related but honestly I have no idea why the midpoint is better than the start or end (or why it doesn't take in a span?  there's no docs on IHostDependentFormattingRuleFactoryService either...)